### ISSUE TITLE
Figure out how not every plugin needs to bundle its own tbl

### ISSUE THREAD

**Issue by distantnative** [distantnative/tbl-for-kirby#12]

**Comment by adamkiss** [comment]
I am not 100% sure, but in theory if you packaged this as an actual Kirby Plugin, you'd in fact register '<tbl-for-kirby>' component during loading - so the plugins could only reference the component in their template section.

The installation can be done via readme method ("don't forget to also install"), or for composer maybe dependency?

**Comment by distantnative** [comment]
@adamkiss yea @sylvainjule and I wanted to avoid the “don’t forget to also install this other plugin otherwise this won’t work and you will complain” text when we initially came up with tbl 

But it needs another solution for sure

**Comment by adamkiss** [comment]
@distantnative Another solution would be to basically merge this and pagetable together and create a "kirby table section", where the query would be configurable - by default it would be pages (= current pagetable plugin, but you could overload it as needed.

**Comment by distantnative** [comment]
@adamkiss That wouldn't work for the prupose where I use tbl: Retour.

**Comment by adamkiss** [comment]
@distantnative Why not? You'd still have the component registered by the table section, you'd just give it a different set of data.

I may be wrong though.

**Comment by distantnative** [comment]
@adamkiss but then Retour must require to also install the pagetable plugin

**Comment by distantnative** [comment]
Plus the question how two different plugins can rely on a shared tbl - while they might require different versions of tbl

**Comment by adamkiss** [comment]
Yeah, I get it. So then, every plugin needs to bundle their own tlb, I don't think there's anything in between

**Comment by distantnative** [comment]
In between could be that plugin add checks and only load dynamically tbl if no other plugin has already loaded it and registered globally.

Or if Kirby at some point supports plugin dependencies.
